docs/CHAIN_CONFIG.md

```diff
@@ -0,0 +1,194 @@
+| Base Sepolia | 84532 | 16 blocks | ~2s | Required (OP Stack) |
+| Arbitrum Sepolia | 421614 | 128 blocks | ~250ms | Required (Arbitrum) |
+| Unichain Sepolia | 1301 | 30 blocks | ~1s | Required (OP Stack) |
+| Berachain Bepolia | 80069 | 15 blocks | ~5s | Not required |
+
+## Usage
+
+### In Deployment Scripts
+
+```solidity
+import { ChainConfig } from "../src/contracts/libraries/ChainConfig.sol";
+
+contract DeployAtlasScript is DeployBaseScript {
+    function run() external {
+        // Get chain-specific configuration
+        ChainConfig.ChainParameters memory chainParams = ChainConfig.getChainParameters(block.chainid);
+        
+        // Use parameters in deployment
+        atlas = new Atlas({
+            escrowDuration: chainParams.escrowDuration,
+            atlasSurchargeRate: chainParams.atlasSurchargeRate,
+            l2GasCalculator: chainParams.l2GasCalculator,
+            // ... other parameters
+        });
+    }
+}
+```
+
+### Helper Functions
+
+```solidity
+// Get chain name
+string memory chainName = ChainConfig.getChainName(block.chainid);
+
+// Check if L2 gas calculator is required
+bool requiresCalc = ChainConfig.requiresL2GasCalculator(block.chainid);
+
+// Get surcharge rates
+ChainConfig.ChainParameters memory params = ChainConfig.getChainParameters(block.chainid);
+uint256 atlasSurcharge = params.atlasSurchargeRate;    // in basis points (1000 = 10%)
+uint256 bundlerSurcharge = params.bundlerSurchargeRate; // in basis points
+```
+
+## Adding New Chains
+
+To add support for a new chain:
+
+1. Add the chain configuration in `ChainConfig.sol`:
+```solidity
+} else if (chainId == YOUR_CHAIN_ID) {
+    return ChainParameters({
+        escrowDuration: BLOCKS,  // Calculate based on block time
+        l2GasCalculator: address(0), // Or specific address if needed
+        atlasSurchargeRate: 1000,  // 10% default
+        bundlerSurchargeRate: 1000, // 10% default
+        name: "YOUR_CHAIN_NAME"
+    });
+}
+```
+
+2. Update `requiresL2GasCalculator()` if the chain needs a gas calculator:
+```solidity
+function requiresL2GasCalculator(uint256 chainId) internal pure returns (bool) {
+    return chainId == 42161 || chainId == 421614 || // Arbitrum chains
+           chainId == 10 || chainId == 11155420 ||    // Optimism chains
+           chainId == 8453 || chainId == 84532 ||     // Base chains
+           chainId == 130 || chainId == 1301 ||       // Unichain chains
+           chainId == YOUR_CHAIN_ID; // Add your chain if needed
+}
+```
+
+## Escrow Duration Calculation
+
+Escrow duration is calculated to achieve approximately 30-35 seconds of real time:
+
+- **Ethereum/Sepolia**: 64 blocks × 12s = ~768s (adjusted for network conditions)
+- **Polygon/Amoy**: 64 blocks × 2s = ~128s (adjusted for faster finality)
+- **OP Stack (Optimism/Base)**: 16 blocks × 2s = ~32s
+- **Arbitrum**: 128 blocks × 250ms = ~32s
+- **Hyperliquid/Unichain**: 30 blocks × 1s = ~30s
```

Comment:
Hyperliquid and Unichain are very different - shouldn't be grouped.

HL = 0.07s block times on their website, but others say 0.2s. Lets assume 0.1s. The escrow period is relevant to unbonding which can be done in small, fast blocks.

Unichain is 250ms blocks I think, and is OP Stack

Reply:
*Unichain is 200ms, same as other OP Stack chains using flashblocks. Im not sure if all blocks are flashblocks though.